caikit_computer_vision/data_model/text_to_image.py

```diff
@@ -0,0 +1,36 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Data structures for text to image."""
+
+
+# Standard
+from typing import List
+
+# Third Party
+from py_to_proto.dataclass_to_proto import Annotated, FieldNumber
+
+# First Party
+from caikit.core import DataObjectBase, dataobject
+from caikit.interfaces.common.data_model import ProducerId
+from caikit.interfaces.vision import data_model as caikit_dm
+import alog
+
+log = alog.use_channel("DATAM")
+
+
+@dataobject(package="caikit_data_model.caikit_computer_vision")
+class TextToImageResult(DataObjectBase):
```

Comment:
I like this more than the current output type! Currently things are written the way they are because of the way software expects image to be formatted, which is basically wrapping encoded bytes of a compressed image. I greatly prefer this also though! For now, I would like to continue with this output format if you're alright with it, since this is what they have been testing with also, but I think it would be a good idea to add this to caikit and update the result to use this in the future.

I suspect this is the route that they will want to go too, since they had already started talking about returning a JSON object with stuff + image instead of just the encoded image 🤞

Reply:
I guess I'm a little confused on how this structure accomplishes the goal of having the output be just encoded bytes. I would expect that you would still need to call some function on the `output` field to get those bytes encoded, right? If this is returned from a `task` in `caikit.runtime`, it would return as a `json` blob or serialized proto with `output` and `producer_id` fields unless there's something in the core `Image` that would provide custom serialization?

Reply:
Yup! That is in the data model for image, the image data model holds an export format, which by default is `png`. when you get the attribute of image data on the image backend, it makes a BytesIO object and exports it with PIL [here](https://github.com/caikit/caikit/blob/main/caikit/interfaces/vision/data_model/backends/image_pil_backend.py#L57)!